test/groovy/util/JenkinsReadYamlRule.groovy

```diff
@@ -25,13 +25,16 @@ class JenkinsReadYamlRule implements TestRule {
         return new Statement() {
             @Override
             void evaluate() throws Throwable {
-                testInstance.helper.registerAllowedMethod("readYaml", [Map], { Map m ->
-                    if(m.text) {
-                        return new Yaml().load(m.text)
-                    } else if(m.file) {
-                        throw new UnsupportedOperationException()
+                testInstance.helper.registerAllowedMethod("readYaml", [Map], { Map parameters ->
+                    if(parameters.text) {
+                        return new Yaml().load(parameters.text)
+                    } else if(parameters.file) {
+                        if (parameters.file == '.pipeline/config.yml') {
+                            return new Yaml().load(new File('test/resources/test_pipeline_config.yml').text)
```

Comment:
The JenkinsReadYamlRule as available in the master should be sufficient.

I don't see a real need for mocking the reading of the config file.
We can mock the configuration as we would like to have it for the test via adding the respective configuration map directly to the `commonPipelineEnvironment` via:

`cpe.configuration = [steps:[mtaBuild:[mtaJarLocation: '/custom/mta']]]`

In the test you don't need to call `setupCommonPipelineEnvironment` anymore but rather bring the `commonPipelineEnvironment` into a state as expected.
An example can be found here:
https://github.com/SAP/jenkins-library/blob/master/test/groovy/ArtifactSetVersionTest.groovy#L102

If you see the need for a custom file this should not be hardcoded but we should rather instantiate the rule with a base path to a location where the test-specific file is stored. I suggest doing it as already done here: 
https://github.com/SAP/jenkins-library/blob/master/test/groovy/util/JenkinsReadFileRule.groovy#L13

Reply:
I would prefer to mock the original configuration file `.pipeline/config.yml`. I think is more readable, and is the original flow, like in https://github.com/SAP/jenkins-library/blob/master/test/groovy/SetupCommonPipelineEnvironmentTest.groovy

Reply:
If you see the need in your case this is fine.

Then I suggest to extend the reading of a custom file with the possibility of defining a test-root.
Here you just should make sure that this is not incompatible for existing tests.